src/test/integration/integration-test.py

```diff
@@ -21,18 +21,54 @@ def setup(cls, config_path, openapi_path):
     def cleanup(cls):
         cls.session.close()
 
-    # Test case: GET /spendingauthority with valid authority onids
     def test_get_authority_onids(self, endpoint='/spendingauthority'):
         for onid in self.test_cases['valid_authority_onids']:
             params = {'onid': onid}
             response = utils.make_request(self, endpoint, 200, params=params)
-            spending_schema = utils.get_resource_schema(
-                self, 'SpendingAuthorityResource'
-            )
-            utils.check_schema(self, response, spending_schema)
-            response_data = response.json()['data']
-            actual_onid = response_data['id']
-            self.assertEqual(actual_onid.lower(), onid.lower())
+            self.checking_response_schema(response, onid)
+            self.checking_spending_limits(response)
+
+    # Test case: GET /spendingauthority with valid authority onids
+    def checking_response_schema(self, response, onid):
+        spending_schema = utils.get_resource_schema(
+            self, 'SpendingAuthorityResource'
+        )
+        utils.check_schema(self, response, spending_schema)
+        response_data = response.json()['data']
```

Comment:
Instead of passing entire `response` object into this function then only fetch the `data` field, you can just send response.json()['data'] in the first place.

Reply:
response is still being used in `utils.check_schema(self, response, spending_schema)`
i can merge `checking_response_schema` with `test_get_authority_onids` in the same function and have it call `checking_spending_limits` with only passing the attributes

Reply:
Exactly!